File: src/main/java/upbrella/be/user/dto/response/KakaoLoginResponse.java
package upbrella.be.user.dto.response;

import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.*;
import upbrella.be.user.dto.request.KakaoAccount;

import java.io.Serializable;

import static com.fasterxml.jackson.databind.PropertyNamingStrategies.*;

@Getter
@Builder
@JsonNaming(SnakeCaseStrategy.class)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class KakaoLoginResponse implements Serializable {

    private Long id;
    private KakaoAccount kakaoAccount;
}

File: src/main/kotlin/upbrella/be/user/dto/response/KakaoLoginResponse.kt
package upbrella.be.user.dto.response

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import upbrella.be.user.dto.request.KakaoAccount
import java.io.Serializable

@JsonNaming(SnakeCaseStrategy::class)
data class KakaoLoginResponse(
    val id: Long? = null,
    val kakaoAccount: KakaoAccount? = null
) : Serializable
